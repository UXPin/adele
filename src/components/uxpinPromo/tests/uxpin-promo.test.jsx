import { mount } from 'enzyme';
import React from 'react';
import 'jest-styled-components';
import UXPinPromo from '../uxpin-promo';
import prototyping from '../../assets/powerful-prototyping.png';
import deck from '../../assets/icon-board.png';

it('renders UXPin promo with image on the left correctly', () => {
  const wrapper = mount(<UXPinPromo
    layout="left"
    image={prototyping}
    header="Evangelize Design System with a Free Template!"
    text="40+ Slides ready for action. Perfect for team and stakeholder presentation.
    Available as a powerpoint and keynote deck."
    buttonLabel="Learn more & download"
    alt="UXPin Desgin System Evangelization Free Presentation Template"
    link="https://www.uxpin.com"
  />);
  expect(wrapper).toMatchSnapshot();
});

it('renders UXPin promo with image on the right correctly', () => {
  const wrapper = mount(<UXPinPromo
    layout="right"
    image={deck}
    header="Use One Platform to Build System, Prototype and Collaborate!"
    text="Join thousands of companies (including PayPal, Sapient and HBO)
    and optimize your design process with UXPin."
    buttonLabel="Start a free trial now!"
    shadow
    alt="UXPin - The Design System Platform"
    link="https://www.uxpin.com"
  />);
  expect(wrapper).toMatchSnapshot();
});
