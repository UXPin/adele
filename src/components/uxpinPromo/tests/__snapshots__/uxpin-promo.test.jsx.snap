// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders UXPin promo with image on the left correctly 1`] = `
.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  height: 40px;
  min-height: 40px;
  width: 100%;
  max-width: 200px;
  background-color: #006cff;
  border-radius: 25px;
  border: 1px solid #006cff;
  font-family: proxima-nova,Proxima Nova,Helvetica Neue,Helvetica,Arial,sans-serif;
  font-size: 16px;
  font-weight: 300;
  color: white;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  -webkit-transition: all ease-in 0.1s;
  transition: all ease-in 0.1s;
}

.c4:hover {
  background-color: #3389ff;
  border-color: #3389ff;
  box-shadow: 4px 4px 20px rgba(179,211,255,0.6);
}

.c4:active {
  background-color: #0061e6;
  box-shadow: 'none';
}

.c4:focus {
  outline: none;
  background-color: #3389ff;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 2 0 0;
  -ms-flex: 2 0 0;
  flex: 2 0 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 800px;
}

.c0:first-child {
  margin-right: 80px;
}

.c2 {
  margin: 40px 0 40px 0;
}

.c1 {
  max-width: 600px;
  font-family: proxima-nova,Proxima Nova,Helvetica Neue,Helvetica,Arial,sans-serif;
  font-size: 42px;
  color: #323335;
  font-weight: 300;
  line-height: calc(42px * 1.3);
  margin-bottom: 20px;
}

.c3 {
  max-width: 600px;
  font-family: proxima-nova,Proxima Nova,Helvetica Neue,Helvetica,Arial,sans-serif;
  font-size: 16px;
  color: #777777;
  font-weight: 300;
  line-height: calc(16px * 1.6);
  padding-bottom: 20px;
}

<UXPinPromo
  alt="UXPin Desgin System Evangelization Free Presentation Template"
  buttonLabel="Learn more & download"
  header="Evangelize Design System with a Free Template!"
  image="test-file-stub"
  layout="left"
  link="https://www.uxpin.com"
  text="40+ Slides ready for action. Perfect for team and stakeholder presentation. Available as a powerpoint and keynote deck."
>
  <styled.section>
    <section
      className="c0"
    >
      <styled.h2>
        <h2
          className="c1"
        >
          Evangelize Design System with a Free Template!
        </h2>
      </styled.h2>
      <styled.img
        alt="UXPin Desgin System Evangelization Free Presentation Template"
        src="test-file-stub"
      >
        <img
          alt="UXPin Desgin System Evangelization Free Presentation Template"
          className="c2"
          src="test-file-stub"
        />
      </styled.img>
      <styled.p>
        <p
          className="c3"
        >
          40+ Slides ready for action. Perfect for team and stakeholder presentation. Available as a powerpoint and keynote deck.
        </p>
      </styled.p>
      <Button
        action="https://www.uxpin.com"
        label="Learn more & download"
        tab={5}
        targetBlank={true}
        type="link"
      >
        <styled.a
          href="https://www.uxpin.com"
          tabIndex={5}
          target="_blank"
        >
          <a
            className="c4"
            href="https://www.uxpin.com"
            tabIndex={5}
            target="_blank"
          >
            Learn more & download
          </a>
        </styled.a>
      </Button>
    </section>
  </styled.section>
</UXPinPromo>
`;

exports[`renders UXPin promo with image on the right correctly 1`] = `
.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  height: 40px;
  min-height: 40px;
  width: 100%;
  max-width: 200px;
  background-color: #006cff;
  border-radius: 25px;
  border: 1px solid #006cff;
  font-family: proxima-nova,Proxima Nova,Helvetica Neue,Helvetica,Arial,sans-serif;
  font-size: 16px;
  font-weight: 300;
  color: white;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  -webkit-transition: all ease-in 0.1s;
  transition: all ease-in 0.1s;
}

.c4:hover {
  background-color: #3389ff;
  border-color: #3389ff;
  box-shadow: 4px 4px 20px rgba(179,211,255,0.6);
}

.c4:active {
  background-color: #0061e6;
  box-shadow: 'none';
}

.c4:focus {
  outline: none;
  background-color: #3389ff;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 2 0 0;
  -ms-flex: 2 0 0;
  flex: 2 0 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 800px;
}

.c0:first-child {
  margin-right: 80px;
}

.c2 {
  margin: 40px 0 40px 0;
}

.c1 {
  max-width: 600px;
  font-family: proxima-nova,Proxima Nova,Helvetica Neue,Helvetica,Arial,sans-serif;
  font-size: 42px;
  color: #323335;
  font-weight: 300;
  line-height: calc(42px * 1.3);
  margin-bottom: 20px;
}

.c3 {
  max-width: 600px;
  font-family: proxima-nova,Proxima Nova,Helvetica Neue,Helvetica,Arial,sans-serif;
  font-size: 16px;
  color: #777777;
  font-weight: 300;
  line-height: calc(16px * 1.6);
  padding-bottom: 20px;
}

<UXPinPromo
  alt="UXPin - The Design System Platform"
  buttonLabel="Start a free trial now!"
  header="Use One Platform to Build System, Prototype and Collaborate!"
  image="test-file-stub"
  layout="right"
  link="https://www.uxpin.com"
  shadow={true}
  text="Join thousands of companies (including PayPal, Sapient and HBO) and optimize your design process with UXPin."
>
  <styled.section>
    <section
      className="c0"
    >
      <styled.h2>
        <h2
          className="c1"
        >
          Use One Platform to Build System, Prototype and Collaborate!
        </h2>
      </styled.h2>
      <styled.img
        alt="UXPin - The Design System Platform"
        src="test-file-stub"
      >
        <img
          alt="UXPin - The Design System Platform"
          className="c2"
          src="test-file-stub"
        />
      </styled.img>
      <styled.p>
        <p
          className="c3"
        >
          Join thousands of companies (including PayPal, Sapient and HBO) and optimize your design process with UXPin.
        </p>
      </styled.p>
      <Button
        action="https://www.uxpin.com"
        label="Start a free trial now!"
        tab={5}
        targetBlank={true}
        type="link"
      >
        <styled.a
          href="https://www.uxpin.com"
          tabIndex={5}
          target="_blank"
        >
          <a
            className="c4"
            href="https://www.uxpin.com"
            tabIndex={5}
            target="_blank"
          >
            Start a free trial now!
          </a>
        </styled.a>
      </Button>
    </section>
  </styled.section>
</UXPinPromo>
`;
